src/screens/AirdropDashboard/components/AirdropInProgress.tsx

```diff
@@ -105,12 +82,91 @@ export const AirdropInProgress: FC<Props> = props => {
         </Text>
       </View>
       <Countdown dataEnd={CONST.airdropDate} />
-      <AirdropInProgressContent {...props} />
+      {loading && (
+        <View style={styles.loadingContainer}>
+          <Loading />
+        </View>
+      )}
+      {error && (
+        <View style={styles.errorContainer}>
+          <Error />
+        </View>
+      )}
+      {!error && !loading && (
+        <>
+          {userHasSubscribedWallets ? (
+            <>
+              <AirdropCarousel
+                styles={styles.carouselStyles}
+                items={getCarouselItems(subscribedWallets, usersQuantity)}
+                setRef={setRef}
+                onItemSnap={onCarouselItemSnap}
+              />
+              {!communityCarouselActive && (
+                <View style={styles.walletsListContainer}>
+                  <AirdropWalletsList
+                    wallets={subscribedWallets}
+                    title={i18n.airdrop.dashboard.registeredWallets}
+                    itemCallToAction={(wallet: Wallet) => (
+                      <RegisteredWalletAction onActionClick={() => setCarouselActiveElement(wallet)} />
+                    )}
+                  />
+                </View>
+              )}
+            </>
+          ) : (
+            <>
+              <Image source={images.airdrop} style={styles.airdropImage} />
+              <Text style={styles.description}>{i18n.airdrop.dashboard.desc2}</Text>
+            </>
+          )}
+          {availableWallets?.length > 0 && !communityCarouselActive && (
+            <View style={styles.walletsListContainer}>
+              <AirdropWalletsList
+                wallets={availableWallets}
+                title={i18n.airdrop.dashboard.availableWallets}
+                itemCallToAction={(wallet: Wallet) => (
+                  <AvailableWalletAction onActionClick={() => subscribeWallet(wallet)} />
+                )}
+              />
+            </View>
+          )}
+          {userHasSubscribedWallets && !communityCarouselActive && (
+            <View style={styles.communitySectionContainer}>
+              <CommunitySection
+                onActionClick={() => {
+                  _carouselRef.snapToItem(subscribedWallets.length, true);
+                }}
+              />
+            </View>
+          )}
+          {communityCarouselActive && (
+            <View style={styles.communitySectionContainer}>
+              {listGoals.length > 0 &&
```

Comment:
is this check `listGoals.length > 0 ` needed? will be here a crash if we map an empty array?

Reply:
done